File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/kit/KitItem.java
package top.likoslupus.cellulosesz.api.kit;

import static top.likoslupus.cellulosesz.api.validation.NumericChecks.requireNonNegative;
import static top.likoslupus.cellulosesz.api.validation.TextChecks.requireNonBlank;

import static java.util.Objects.requireNonNull;

/** Lossless inventory stack and its original player-inventory slot. */
public record KitItem(
        int slot,
        String stack
) {

    public KitItem {
        requireNonNegative(slot, "slot");
        stack = requireNonBlank(requireNonNull(stack, "stack").trim(), "stack");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/kit/KitItem.kt
package top.likoslupus.cellulosesz.api.kit

import top.likoslupus.cellulosesz.api.validation.requireNonBlank
import top.likoslupus.cellulosesz.api.validation.requireNonNegative

/** Lossless inventory stack and its original player-inventory slot. */
@JvmRecord
public data class KitItem(
    public val slot: Int,
    public val stack: String
) {

    init {
        slot.requireNonNegative { "slot" }
        stack.requireNonBlank { "stack" }
    }

}
